datafusion/physical-plan/src/stream.rs

```diff
@@ -38,6 +38,113 @@ use tokio::task::JoinSet;
 use super::metrics::BaselineMetrics;
 use super::{ExecutionPlan, RecordBatchStream, SendableRecordBatchStream};
 
+/// Creates a stream from a collection of producing tasks, routing panics to the stream
+pub(crate) struct ReceiverStreamBuilder<O> {
```

Comment:
I made this pub(crate) but it could easily be made public should we wish to do so